Automattic/woocommerce-payments
# feat: disable WCPay confirmation dialog

Fixes https://github.com/Automattic/woocommerce-payments/issues/1564

#### Changes proposed in this Pull Request









Added a modal to display when the merchant attempts to disable WCPay.
Solution for the event listener has been proposed by @superdav42 here: https://github.com/Automattic/woocommerce-payments/issues/1564#issuecomment-837473858

#### Testing instructions





- Ensure you have the latest version of the dev tools: `cd docker/wordpress/wp-content/plugins/woocommerce-payments-dev-tools && git checkout trunk && git pull`
- Ensure you have the "grouped settings" flag enabled in WCPay Dev Tools
- (optional) Ensure you have the "Add UPE additional payment methods" flag enabled in WCPay Dev Tools
- Go to the list of gateways: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section
- Attempting to disable WCPay shows the modal
- The modal is not displayed when enabling WCPay
- Cancelling the action on the modal does not disable WCPay
- Confirming the action on the modal disables WCPay

-------------------
